nuget/helpers/lib/NuGetUpdater/NuGetUpdater.Core/Discover/DiscoveryWorker.cs

```diff
@@ -225,12 +229,23 @@ private static ImmutableArray<string> ExpandEntryPointsIntoProjects(IEnumerable<
             if (seenProjects.Add(candidateEntryPoint))
             {
                 string extension = Path.GetExtension(candidateEntryPoint).ToLowerInvariant();
-                if (extension == ".sln")
+                if (extension is ".sln" or ".slnx")
```

Comment:
I strongly suggest only using the new parser library for .sln files for the near future. It it much more stringent than the SolutionFile.Parse method and will likely cause unnecessary failures for you. We stopped using it in MSBuild for this reason. We're working with the vs-solutionpersistence team to make sure their sln parsing works with more real-world data sets.

Reply:
Agreed. This exactly what is done here.

Reply:
So keeping the current `sln` support as-is and just adding `slnx` additively through SolutionPersistence right?

Reply:
Yes, exactly

Reply:
Updated the PR, indeed this is the safest approach to add `slnx` support without breaking the existing.